src/FsAutoComplete.Core/FileSystem.fs

```diff
@@ -824,22 +824,22 @@ type FileSystem(actualFs: IFileSystem, tryFindFile: string<LocalPath> -> Volatil
     member _.IsPathRootedShim(p: string) =
       let r = isWindowsStyleRootedPath p || isUnixStyleRootedPath p
 
-      fsLogger.debug (
-        Log.setMessage "Is {path} rooted? {result}"
-        >> Log.addContext "path" p
-        >> Log.addContext "result" r
-      )
+      // fsLogger.debug (
+      //   Log.setMessage "Is {path} rooted? {result}"
+      //   >> Log.addContext "path" p
+      //   >> Log.addContext "result" r
+      // )
 
       r
 
     member _.GetFullPathShim(f: string) =
       let expanded = Path.FilePathToUri f |> Path.FileUriToLocalPath
 
-      fsLogger.debug (
-        Log.setMessage "{path} expanded to {expanded}"
-        >> Log.addContext "path" f
-        >> Log.addContext "expanded" expanded
-      )
+      // fsLogger.debug (
+      //   Log.setMessage "{path} expanded to {expanded}"
+      //   >> Log.addContext "path" f
+      //   >> Log.addContext "expanded" expanded
+      // )
```

Comment:
```fsharp
let inline debug a b c = ()

let foo () =
  debug 1 (2+2) 4
```

it seems the compiler is optimizing this, wondering if this can be used to make the debug function a no-op in release builds while keeping the code around for debug builds?

Reply:
I guess no:

```fsharp
let inline debug a b c = ()

let foo () =
  debug (new obj()) (new obj()) (new obj())
```

it doesn't get elided.

Reply:
could always ifdef them i guess